src/simperium/queues/local-queue.js

```diff
@@ -0,0 +1,295 @@
+// @flow
+import events from 'events';
+import * as change_util from '../util/change';
+import type { BucketOperation } from '../util/change';
+
+const { EventEmitter } = events;
+
+/**
+ * A ghost represents a version of a bucket object as known by Simperium
+ *
+ * Generally a client will keep the last known ghost stored locally for efficient
+ * diffing and patching of Simperium change operations.
+ *
+ * @typedef {Object} Ghost
+ * @property {Number} version - the ghost's version
+ * @property {String} key - the simperium bucket object id this ghost is for
+ * @property {Object} data - the data for the given ghost version
+ */
+type Ghost = { version: number, key: string, data: {} };
+
+/**
+ * A GhostStore provides the store mechanism for ghost data that the Channel
+ * uses to maintain syncing state and producing change operations for
+ * Bucket objects.
+ *
+ * @interface GhostStore
+ */
+interface GhostStore {
+
+	/**
+	 * Callback function used by the ghost store to iterate over existing ghosts
+	 *
+	 * @callback ghostIterator
+	 * @param {Ghost} - the current ghost
+	 */
+
+	/**
+	 * Retrieve a Ghost for the given bucket object id
+	 *
+	 * @function
+	 * @name GhostStore#get
+	 * @param {String} id - bucket object id
+	 * @returns {Promise<Ghost>} - the ghost for this object
+	 */
+	get( id: string ): Promise<Ghost>;
+
+	/**
+	 * Save a ghost in the store.
+	 *
+	 * @function
+	 * @name GhostStore#put
+	 * @param {String} id - bucket object id
+	 * @param {Number} version - version of ghost data
+	 * @param {Object} data - object literal to save as this ghost's data for this version
+	 * @returns {Promise<Ghost>} - the ghost for this object
+	 */
```

Comment:
what's going on here? are these missing or is the JSDoc comment supposed to indicate that the functions exist?

Reply:
The functions appear to be missing. I think this is the only Flow typed file that tries to use a `GhostStore` and the only method it needs is `GhostStore.get`.

Once this `interface` is moved to the proper location (a Flow typed `Bucket` or `Channel` module perhaps) the missing interface methods would have failed the type checker.

Reply:
Ha, I event wrote as much over a year ago:

> The GhostStore type is currently located in the LocalQueue module because it is the only one using that type definition at this moment. It will need a better home when the Channel module gets flowtyped but I think we can cross that bridge when we tackle the Channel changes.